olp-cpp-sdk-dataservice-read/src/ApiResponseWithNetworkStats.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <olp/core/client/ApiResponse.h>
+#include <olp/core/client/HttpResponse.h>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+template <typename Result, typename Error>
+class ApiResponseWithNetworkStats : public client::ApiResponse<Result, Error> {
```

Comment:
How about `ExtendedApiResponse` and make it more generic with an additional template parameter instead of fixed to NetworkStatistics? Might be more flexible for the future.